### PR TITLE
Automatically adjust relative paths

### LINKED ISSUE #55: Use better location as base for relative paths
When trying to load a SQLite database from a relative path (`Content\Db\Assets.db`) it will try to load the file from the current process location which is klr.exe instead of the location of the working directory.

**Workarounds**

* Use an absolute path
* Use `PathResolver.ResolveAppRelativePath` from the Configuration project.

### PR DESCRIPTION
**Pull request by natemcmaster**
Always adjust relative paths to the ApplicationBasePath.

Fix #55

### CODE REVIEW

File: src/Microsoft.Data.Sqlite/SqliteConnection.cs [hunk 1]
```diff
@@ -96,61 +98,88 @@ public override void Open()
...
+            {
+                path = AdjustForRelativeDirectory(path);
+            }
+            
             var rc = NativeMethods.sqlite3_open_v2(path, out _db, flags, vfs: null);
             MarshalEx.ThrowExceptionForRC(rc, _db);
             SetState(ConnectionState.Open);
-
+            
             SetFolders();
```
Comment (bricelam) [inline src/Microsoft.Data.Sqlite/SqliteConnection.cs]: A `partial` method could be used here since it's a no-op outside of netcore50.

**Comment by natemcmaster** [comment]
Rebased and updated.

File: src/Microsoft.Data.Sqlite/SqliteConnection.cs [hunk 2]
```diff
@@ -105,52 +111,76 @@ public override void Open()
...
-            var appData = (dynamic)appDataType?.GetTypeInfo()
-                .GetDeclaredProperty("Current").GetMethod.Invoke(null, null);
-            return appData;
+            try
+            {
+                return Path.GetFullPath(Path.Combine(DefaultBasePath, path));
+            }
+            catch (Exception ex)
+            {
+                if (ex is ArgumentException || ex is NotSupportedException)
```
Comment (bricelam) [inline src/Microsoft.Data.Sqlite/SqliteConnection.cs]: Could use C# 6 `catch...when` syntax.

Comment (bricelam) [inline src/Microsoft.Data.Sqlite/SqliteConnection.cs]: Or `where` or `if`... whatever it ended up being called :smile:

File: src/Microsoft.Data.Sqlite/SqliteConnection.cs [hunk 3]
```diff
@@ -105,52 +111,76 @@ public override void Open()
...
+                return path;
+            }
 
-            if (appData == null)
+            if (path.StartsWith("file:"))
             {
-                return;
+#if NETCORE50
+                // UWP cannot adjust for URI on UWP
+                return path;
```
Comment (bricelam) [inline src/Microsoft.Data.Sqlite/SqliteConnection.cs]: If it's absolute, will it work? If so, we could support this by parsing it using the `Uri` class and fixing up the path.

**Comment by bricelam** [comment]
:shipit:

File: src/Microsoft.Data.Sqlite/SqliteConnection.cs (hunk 3 above)
Comment (natemcmaster) [inline src/Microsoft.Data.Sqlite/SqliteConnection.cs]: I tried this at first. Even with an absolute path, Uri file paths do not work on UWP.

File: src/Microsoft.Data.Sqlite/SqliteConnection.cs (hunk 2 above)
Comment (natemcmaster) [inline src/Microsoft.Data.Sqlite/SqliteConnection.cs]: C#6 ftw.

File: src/Microsoft.Data.Sqlite/SqliteConnection.cs (hunk 3 above)
Comment (bricelam) [inline src/Microsoft.Data.Sqlite/SqliteConnection.cs]: I think you'd need to call `sqlite3_config(SQLITE_CONFIG_URI, 1)` first.

Comment (bricelam) [inline src/Microsoft.Data.Sqlite/SqliteConnection.cs]: But it seems pretty low priority...

Comment (natemcmaster) [inline src/Microsoft.Data.Sqlite/SqliteConnection.cs]: The only a reason a user would need Uri syntax is for [query parameters](http://www.sqlite.org/uri.html#coreqp) we don't support. We could add this to the connection string build itself if needed, but for now I agree--this is low priority.

Comment (bricelam) [inline src/Microsoft.Data.Sqlite/SqliteConnection.cs]: Named, shared-cache, in-memory databases are the only scenarios I find interesting here. :smile_cat:

**Pull request merged**